airbrakes/hardware/imu.py

```diff
@@ -123,69 +184,27 @@ def _query_imu_for_data_packets(self, port: str) -> None:
             self._fetch_data_loop(port)
 
     def _fetch_data_loop(self, port: str) -> None:
-        # Connect to the IMU
-        connection = mscl.Connection.Serial(port)
-        node = mscl.InertialNode(connection)
-        timeout = 10
-
+        """
+        Continuously fetch data packets from the IMU and process them.
+        :param port: The serial port to connect to the IMU.
+        """
         while self._running.value:
-            # Get the latest data packets from the IMU, with the help of `getDataPackets`.
-            # `getDataPackets` accepts a timeout in milliseconds.
-            # During IMU configuration (outside of this code), we set the sampling rate of the IMU
-            # as 1ms for RawDataPackets, and 2ms for EstimatedDataPackets.
-            # So we use a timeout of 10ms which should be more
-            # than enough. If the timeout is hit, the function will return an empty list.
-
-            packets: mscl.MipDataPackets = node.getDataPackets(timeout)
-            # Every loop iteration we get the latest data in form of packets from the imu
+            # Reconnect to the IMU at the start of each loop iteration.
+            connection = mscl.Connection.Serial(port)
+            node = mscl.InertialNode(connection)
```

Comment:
I have no idea what you said since you deleted it, but why reconnect?

Reply:
it was me being dumb basically